crates/ruff_python_ast/src/comparable.rs

```diff
@@ -326,6 +326,23 @@ impl<'a> From<&'a ast::Decorator> for ComparableDecorator<'a> {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum ComparableSingleton {
```

Comment:
Nit: I find `Singleton` a confusing name. I assume it comes from Python's AST definition?

Or should this match the literal case? https://peps.python.org/pep-0622/#literal-patterns

Reply:
> I assume it comes from Python's AST definition?

Yes, and in turn ours as both uses the "Singleton" suffix.

It is for the literal pattern but only a subset of it - `None`, `True`, `False`. The string and number literal are represented using `PatternMatchValue` while the singletons are using `PatternMatchSingleton`.

https://play.ruff.rs/2834cfa6-4d0d-4aaf-bbd2-1bd722fdd2ba